lib/faraday/utils.rb

```diff
@@ -95,7 +95,14 @@ def to_hash() ::Hash.new.update(self) end
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Find the last set of response headers.
+        start_index = headers.rindex { |x| x.match(/^HTTP\//) }
```

Comment:
@iMacTia Hey, I ran into this as well. Agree this is a VCR bug but since the latest release (3.0.3) was last year I don't expect a quick fix. 
Would it be possible to add a `.to_i` cast here to keep it compatible?
For now the easiest solution would be to downgrade faraday to "0.12.1"